src/autocomplete.js

```diff
@@ -611,20 +619,33 @@ class CompletionProvider {
             // TODO add support for options.deleteSuffix
             if (!this.completions)
                 return false;
-            if (this.completions.filterText) {
+            
+            var replaceBefore = this.completions.filterText.length;
+            var replaceAfter = 0;
+            if (data.range && data.range.start.row === data.range.end.row) {
+                replaceBefore -= this.initialPosition.prefix.length;
+                replaceBefore += this.initialPosition.pos.column - data.range.start.column;
+                replaceAfter += data.range.end.column - this.initialPosition.pos.column;
+            }
+
+            if (replaceBefore || replaceAfter) {
                 var ranges;
                 if (editor.selection.getAllRanges) {
                     ranges = editor.selection.getAllRanges();
-                } else {
+                }
+                else {
                     ranges = [editor.getSelectionRange()];
                 }
                 for (var i = 0, range; range = ranges[i]; i++) {
-                    range.start.column -= this.completions.filterText.length;
+                    range.start.column -= replaceBefore;
+                    range.end.column += replaceAfter;
                     editor.session.remove(range);
                 }
             }
-            if (data.snippet)
-                snippetManager.insertSnippet(editor, data.snippet, {range: data.range});
+          
+            if (data.snippet) {
+                snippetManager.insertSnippet(editor, data.snippet, {range: undefined});
```

Comment:
Can you help me understand what is happening here, like why is it better if the range is undefined all of a sudden?

Reply:
This shouldn't be there, I forgot to cleanup this after testing 👍